pkg/util/downloader_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/minikube/tests"
+)
+
+func TestGetISOFileURI(t *testing.T) {
+	dler := DefaultDownloader{}
+
+	tests := map[string]string{
+		"file:///test/path/minikube-test.iso":                           "file:///test/path/minikube-test.iso",
+		"https://storage.googleapis.com/minikube/iso/minikube-test.iso": "file://" + filepath.ToSlash(filepath.Join(constants.GetMinipath(), "cache", "iso", "minikube-test.iso")),
+	}
+
+	for input, expected := range tests {
+		if isoFileURI := dler.GetISOFileURI(input); isoFileURI != expected {
+			t.Fatalf("Expected GetISOFileURI with input %s to return %s but instead got: %s", input, expected, isoFileURI)
+		}
+	}
+
+}
+
+var testISOString = "hello"
+
+type ISOHandlerCorrect struct{}
+
+func (h *ISOHandlerCorrect) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could you move this declaration into the test that uses it? Something like: https://golang.org/pkg/net/http/httptest/#example_Server

Same for the other structs.